libraries/dawg-client/src/main/java/com/comcast/video/dawg/cats/ir/IrClient.java

```diff
@@ -59,14 +60,18 @@
     private String port;
     private String remoteType;
     private String catsKeySetMapping;
+    private boolean proxyEnabled;
 
     /**
      * Creates a client for the stb with the given metadata
      * @param irMeta The metadata about the stb
      */
     public IrClient(IrMeta irMeta) {
         Validate.notNull(irMeta);
-        this.catsHost = irMeta.getCatsServerHost();
+        MetaStb stb = (MetaStb) irMeta;
```

Comment:
You shouldn't cast this to MetaStb. The point of making an interface IrMeta that MetaStb implements is we only need to pass in the necessary data to IrClient, it can be any data source.
If proxyEnabled and rackProxyUrl are properties that are necessary for IR, then they belong in IrMeta.
I guess we didn't do the same for PowerClient though.